src/response.rs

```diff
@@ -522,7 +519,7 @@ impl<'a, 'b> Response<'a, 'b> {
         let mime = path
             .extension()
             .and_then(|ext| to_mime(&ext.to_string_lossy()))
-            .unwrap_or(Mime(TopLevel::Application, SubLevel::Ext("octet-stream".into()), vec![]));
+            .unwrap_or_else(|| Mime(TopLevel::Application, SubLevel::Ext("octet-stream".into()), vec![]));
```

Comment:
I’m unsure what you mean here. If the `Mime(…)` value is to be unused, the closure won’t be called and the string won’t be allocated. And `unwrap_or_else` is `#[inline]`d, so I’d expect the closure itself to have no-cost.